sql/parser/builtins.go

```diff
@@ -943,6 +961,125 @@ func datumToString(datum Datum) (string, error) {
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
 
+var replaceSubRe = regexp.MustCompile(`\\[&1-9]`)
+
+func regexpReplace(s, pattern, to, sqlFlags string) (Datum, error) {
+	pattern, global, err := regexpEvalFlags(pattern, sqlFlags)
+	if err != nil {
+		return nil, err
+	}
+
+	patternRe, err := regexp.Compile(pattern)
```

Comment:
Yeah, we probably need to verify that the cached regex has the same pattern for `LIKE` and `SIMILAR TO`. Ugh. Certainly fixable, though perhaps an LRU cache would be better than the current approach.